TypedSyntax/src/node.jl

```diff
@@ -306,15 +307,22 @@ end
 
 function getsrc(@nospecialize(f), @nospecialize(t))
     srcrts = code_typed(f, t; debuginfo=:source, optimize=false)
-    return only(srcrts)
+    src, rt = only(srcrts)
+    if hasfield(typeof(src), :parent) && src.parent !== nothing
+        mi = src.parent
+    else
+        mi = Base.method_instance(f, t)
+    end
+    return src, rt, mi
```

Comment:
Also, it seems to be confusing that methods with the same name return different type objects. I would like to extract `MethodInstance` lookup logic outside of this function rather than including it here.